scm/driver/gitlab/webhook.go

```diff
@@ -904,4 +1092,139 @@ type (
 			Homepage    string `json:"homepage"`
 		} `json:"repository"`
 	}
+
+	pipelineHook struct {
+		ObjectKind       string           `json:"object_kind"`
+		ObjectAttributes ObjectAttributes `json:"object_attributes"`
+		MergeRequest     MergeRequest     `json:"merge_request"`
+		User             user             `json:"user"`
+		Project          Project          `json:"project"`
+		Commit           Commit           `json:"commit"`
+		SourcePipeline   SourcePipeline   `json:"source_pipeline"`
+		Builds           []Build          `json:"builds"`
+	}
+
+	ObjectAttributes struct {
+		ID         int        `json:"id"`
+		IID        int        `json:"iid"`
+		Name       string     `json:"name"`
+		Ref        string     `json:"ref"`
+		Tag        bool       `json:"tag"`
+		SHA        string     `json:"sha"`
+		BeforeSHA  string     `json:"before_sha"`
+		Source     string     `json:"source"`
+		Status     string     `json:"status"`
+		Stages     []string   `json:"stages"`
+		CreatedAt  string     `json:"created_at"`
+		FinishedAt string     `json:"finished_at"`
+		Duration   int        `json:"duration"`
+		Variables  []Variable `json:"variables"`
+		URL        string     `json:"url"`
+	}
+
+	Variable struct {
+		Key   string `json:"key"`
+		Value string `json:"value"`
+	}
+
+	MergeRequest struct {
+		ID                  int    `json:"id"`
```

Comment:
many vendors do not use numeric `id` which means this abstraction may break when we try to add bitbucket or others

Reply:
done